java/deepgram_audio_summary/Index.java

```diff
@@ -0,0 +1,171 @@
+import java.util.Collections;
+import java.util.stream.Collectors;
+import java.util.*;
+import java.util.stream.*;
+import java.net.HttpURLConnection;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+
+import com.google.gson.Gson;
+import org.apache.commons.validator.routines.UrlValidator;
+
+final Gson gson=new Gson();
+
+public RuntimeResponse main(RuntimeRequest req,RuntimeResponse res)throws Exception{
+        // Validate that values present in the request are not empty (payload, variables)
+        RuntimeResponse errorResponse=checkEmptyPayloadAndVariables(req,res);
+        if(errorResponse!=null){
+            return errorResponse;
+        }
+
+        // Validate the requested payload (URL)
+        String payloadString=req.getPayload().toString();
+        Map<String, Object> payload=gson.fromJson(payloadString,Map.class);
```

Comment:
If an invalid payload is supplied, no response is returned:

```shell
$ curl -H "X-Internal-Challenge: secret-key" \
>     -H "Content-Type: application/json" \
>     -X POST http://localhost:3000/ \
>     -d '{"variables": {"DEEPGRAM_SECRET_KEY": "asdf"}, "payload": "{"}' | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2391  100  2329  100    62   324k   8857 --:--:-- --:--:-- --:--:--  333k
{
  "stdout": "",
  "stderr": "com.google.gson.JsonSyntaxException: java.io.EOFException: End of input at line 1 column 2 path $.\n\tat com.google.gson.Gson.fromJson(Gson.java:1001)\n\tat com.google.gson.Gson.fromJson(Gson.java:956)\n\tat com.google.gson.Gson.fromJson(Gson.java:905)\n\tat com.google.gson.Gson.fromJson(Gson.java:876)\n\tat io.openruntimes.java.Wrapper.main(Wrapper.java:29)\n\tat io.openruntimes.java.Server.execute(Server.java:43)\n\tat org.rapidoid.http.handler.optimized.DelegatingParamsAwareReqRespHandler.handleReq(DelegatingParamsAwareReqRespHandler.java:49)\n\tat org.rapidoid.http.handler.AbstractHttpHandlerDecorator.handleReqAndPostProcess(AbstractHttpHandlerDecorator.java:48)\n\tat org.rapidoid.http.handler.HttpManagedHandlerDecorator$3.invokeNext(HttpManagedHandlerDecorator.java:161)\n\tat org.rapidoid.http.handler.HttpManagedHandlerDecorator$3.invoke(HttpManagedHandlerDecorator.java:121)\n\tat org.rapidoid.http.handler.HttpAuthWrapper.wrap(HttpAuthWrapper.java:70)\n\tat org.rapidoid.http.handler.HttpManagedHandlerDecorator.wrap(HttpManagedHandlerDecorator.java:185)\n\tat org.rapidoid.http.handler.HttpManagedHandlerDecorator.handleWithWrappers(HttpManagedHandlerDecorator.java:100)\n\tat org.rapidoid.http.handler.HttpManagedHandlerDecorator.access$200(HttpManagedHandlerDecorator.java:39)\n\tat org.rapidoid.http.handler.HttpManagedHandlerDecorator$2.run(HttpManagedHandlerDecorator.java:83)\n\tat org.rapidoid.job.PredefinedContextJobWrapper.run(PredefinedContextJobWrapper.java:56)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.base/java.lang.Thread.run(Thread.java:829)\nCaused by: java.io.EOFException: End of input at line 1 column 2 path $.\n\tat com.google.gson.stream.JsonReader.nextNonWhitespace(JsonReader.java:1396)\n\tat com.google.gson.stream.JsonReader.doPeek(JsonReader.java:494)\n\tat com.google.gson.stream.JsonReader.hasNext(JsonReader.java:414)\n\tat com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:184)\n\tat com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:145)\n\tat com.google.gson.Gson.fromJson(Gson.java:991)\n\t... 18 more\n"
}
```

Would you please update the code to provide a helpful response?

Reply:
implemented a more user friendly message :)